phpunit/GutenbergTestSuiteLoader.php

```diff
@@ -0,0 +1,119 @@
+<?php declare(strict_types=1);
+/*
+ * This file is part of PHPUnit.
+ *
+ * (c) Sebastian Bergmann <sebastian@phpunit.de>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+namespace phpunit;
+
+use PHPUnit\Framework\TestCase;
+use PHPUnit\Util\FileLoader;
+use PHPUnit\Runner\TestSuiteLoader;
+
+use \Exception;
+use \ReflectionClass;
+use \ReflectionException;
+
+/**
+ * @internal This class is not covered by the backward compatibility promise for PHPUnit
+ *
+ * @deprecated see https://github.com/sebastianbergmann/phpunit/issues/4039
+ */
+class GutenbergTestSuiteLoader implements TestSuiteLoader
+{
+	/**
+	 * @throws Exception
+	 */
+	public function load(string $suiteClassFile): ReflectionClass
```

Comment:
will this break in CI for older PHP versions that don't support these type annotations, or these primitive-type type annotations?

Reply:
Oh no, I think it will! At the same time, removing them breaks the newer PHP versions:

```
Fatal error: Declaration of phpunit\GutenbergTestSuiteLoader::load(string $suiteClassFile) must be compatible with PHPUnit\Runner\TestSuiteLoader::load(string $suiteClassFile): ReflectionClass 
```

That's quite a pickle 🤔 I can only think of using two loader files and choosing the one suited to the current PHP version.

Reply:
note that PHP has long-supported type annotations for classes, while primitive types are a recent addition

```php
public function load( $suiteClassFile ): ReflectionClass
```

Reply:
Wait, does that mean PHPUnit9 won't run on these older PHP versions? If so, we might have to stick with one of the previous releases.

Reply:
I'm saying that it's long been okay to add type annotations in PHP _if those annotations are for classes, arrays, or other complicated types_. It's specifically the `string` and other primitive data types that break on old version, so the snippet I wrote above _with_ the `ReflectionClass` type but _without_ the `string` annotation _should_ work (or maybe the return-type annotation was also a later addition? can't remember).

If PHPUnit9 comes with `function go( int $code, string $message )` then it will probably fail on older versions of PHP, yes. Would be worth the try to confirm.

Reply:
Actually, I don't think we run any CI tests on the older PHP versions at the moment in the Guteneberg repository. It only happens the WordPress-develop repo. @dmsnell can you think of any reason not to go with a PHP 5-incompatible loader?

Reply:
I just confirmed that https://github.com/sebastianbergmann/phpunit/blob/4ddc038c57fd4a51926fc1faa99ff85ad0fed5c0/src/Runner/TestSuiteLoader.php crashes on PHP 7.3 or lower. This means our PHPUnit9 tests already fail on lower PHP versions and this PR doesn't change anything.